pkg/apis/resourcemanager/v1alpha1/vendor_types.go

```diff
@@ -0,0 +1,201 @@
+	VendorStatusArchived VendorStatusValue = "archived"
+)
+
+// CorporationType defines the type of corporation
+// This should reference a valid corporation type from CorporationTypeConfig
+// +kubebuilder:validation:Pattern=^[a-z0-9-]+$
+type CorporationType string
+
+// TaxIdType defines the type of tax identification
+// +kubebuilder:validation:Enum=SSN;EIN;ITIN;UNSPECIFIED
+type TaxIdType string
+
+const (
+	TaxIdTypeSSN         TaxIdType = "SSN"
+	TaxIdTypeEIN         TaxIdType = "EIN"
+	TaxIdTypeITIN        TaxIdType = "ITIN"
+	TaxIdTypeUnspecified TaxIdType = "UNSPECIFIED"
+)
+
+// Address represents a physical address
+type Address struct {
+	// Street address line 1
+	// +kubebuilder:validation:Required
+	Street string `json:"street"`
+
+	// Street address line 2 (optional)
+	// +optional
+	Street2 string `json:"street2,omitempty"`
+
+	// City
+	// +kubebuilder:validation:Required
+	City string `json:"city"`
+
+	// State or province
+	// +kubebuilder:validation:Required
+	State string `json:"state"`
+
+	// Postal or ZIP code
+	// +kubebuilder:validation:Required
+	PostalCode string `json:"postalCode"`
+
+	// Country
+	// +kubebuilder:validation:Required
+	Country string `json:"country"`
+}
+
+// TaxInfo represents tax-related information
+type TaxInfo struct {
+	// Type of tax identification
+	// +kubebuilder:validation:Required
+	TaxIdType TaxIdType `json:"taxIdType"`
+
+	// Tax identification number
+	// +kubebuilder:validation:Required
+	TaxId string `json:"taxId"`
+
+	// Country for tax purposes
+	// +kubebuilder:validation:Required
+	Country string `json:"country"`
+
+	// Tax document reference (e.g., W-9, W-8BEN)
+	// +kubebuilder:validation:Required
+	TaxDocument string `json:"taxDocument"`
+
+	// Whether tax information has been verified
+	// +kubebuilder:default=false
+	TaxVerified bool `json:"taxVerified"`
+
+	// Timestamp of tax verification
+	// +optional
+	VerificationTimestamp *metav1.Time `json:"verificationTimestamp,omitempty"`
+}
+
+// VendorSpec defines the desired state of Vendor
+// +k8s:protobuf=true
+type VendorSpec struct {
+	// Profile type - person or business
+	// +kubebuilder:validation:Required
+	ProfileType VendorProfileType `json:"profileType"`
+
+	// Legal name of the vendor (required)
+	// +kubebuilder:validation:Required
+	LegalName string `json:"legalName"`
+
+	// Nickname or display name
+	// +optional
+	Nickname string `json:"nickname,omitempty"`
+
+	// Billing address
+	// +kubebuilder:validation:Required
+	BillingAddress Address `json:"billingAddress"`
+
+	// Mailing address (if different from billing)
+	// +optional
+	MailingAddress *Address `json:"mailingAddress,omitempty"`
+
+	// Description of the vendor
+	// +optional
+	Description string `json:"description,omitempty"`
+
+	// Website URL
+	// +optional
+	Website string `json:"website,omitempty"`
+
+	// Current status of the vendor
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default=pending
+	Status VendorStatusValue `json:"status"`
```

Comment:
We should use status conditions in the `.status` of the vendor to communicate status. Additional fields could be added for things like verification status, if necessary.